### PR TITLE
host name & value updated with subnet

### PR DESCRIPTION
**Pull request by amolpati30**
### Problem Statement
Update in parameter name and value from subnet component updates 
the parameter in host inheriting that subnet

### Solution
Updated changes in subnet name & value reflects in host inheriting that subnet

### CODE REVIEW

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 5808
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_subnet.py::test_positive_update_subnet_parameter_host_impact
Test Result : ================== 1 passed, 11 warnings in 738.42s (0:12:18) ==================
```

**Comment by ogajduse** [comment]
@amolpati30 CI/CQ is not passing, can you please resolve it?

**Review by lhellebr — APPROVED** [review]
Fix CQ, otherwise LGTM

**Review by ogajduse — CHANGES_REQUESTED** [review]
Since the expected results mention `The inherited subnet parameter in host enc should have updated name and value`, the test should perform some checks against `hammer host enc-dump`.
I am not familiar with ENC, but I bet that someone from the Rocket team (owner of the Puppet component) will help you with it.

If this step is not valid / is not needed, you can also remove it from the expected results.

File: tests/foreman/api/test_subnet.py [hunk 1]
```diff
@@ -354,11 +355,26 @@ def test_positive_update_subnet_parameter_host_impact():
...
-            updated name and value
 
     :BZ: 1470014
     """
+    parameter = [{'name': gen_string('alpha'), 'value': gen_string('alpha')}]
+    org = target_sat.api.Organization().create()
+    loc = target_sat.api.Location(organization=[org]).create()
+    org_subnet = target_sat.api.Subnet(
+        location=[loc], organization=[org], subnet_parameters_attributes=parameter
+    ).create()
```
Comment (shubhamsg199) [inline tests/foreman/api/test_subnet.py:366]: you can have assertions after the first subnet creation to verify if the parameter name and value were set correctly

**Review by shubhamsg199 — APPROVED** [review]
Looks good, one suggestion

File: tests/foreman/api/test_subnet.py (hunk 1 above)
Comment (amolpati30) [reply tests/foreman/api/test_subnet.py:366]: thanks for your suggestion. assertion point is added.

**Review by ogajduse — APPROVED** [review]

**Pull request merged**